### PR TITLE
Feature/add nft 40029527

### PR DESCRIPTION
**Pull request by CrazyOptimist**
Implemented Add NFT UI - column 7 in figma

Added react-hook-form and yup to work with forms, added common/components/Form with Input, TextArea and others to work with react-hook-form

Added common/components/SelectMultiple

### Important changes

Added tailwind JIT mode, so now instead of bloating tailwind config with pixels we can use a class like `w-[123px]`

Updated typescript to 4.3.2 because older version was incompatible with react-hook-form.

### CODE REVIEW

**Review by ngvtuan — COMMENTED** [review]
errors
ui issue 1
ui issue 2
ui issue 3
ui issue 4

File: src/common/components/Icons/Dots.tsx [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-export default function Dots({
+export const Dots: React.FC<TDotsProps> = ({
   size,
   className,
   vertical = false,
-}: TDotsProps): JSX.Element {
+}) => {
   const style = vertical === true ? { transform: 'rotate(90deg)' } : undefined
 
   return (
```
Comment (reacteer) [inline src/common/components/Icons/Dots.tsx]: Let's use Tailwind here.

File: src/features/nft/addNFT/previewStep/PreviewStep.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                    className='absolute top-0 left-0 bottom-0 rounded bg-green-62'
+                    style={{ width: '65%' }}
+                  />
+                </div>
+                <div className='text-xs text-gray-71'>
+                  65% of average Pastel NFT size
+                </div>
+              </div>
+            </div>
+          </div>
```
Comment (reacteer) [inline src/features/nft/addNFT/previewStep/PreviewStep.tsx]: Can we use dummy data here? This is non-blocker.

File: src/common/components/AnimatedModal/AnimatedModal.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  useKey('Escape', () => easyToClose && onClose?.())
+
+  const setModalRef = (el: HTMLDivElement | null) => {
+    modalRef.current = el || undefined
+    if (el) {
+      setLocalOpen(true)
+      el.getBoundingClientRect()
+      el.classList.add('opacity-100')
+    }
+  }
```
Comment (reacteer) [inline src/common/components/AnimatedModal/AnimatedModal.tsx]: We can use Tailwind here.

File: src/features/nft/addNFT/previewStep/PreviewStep.tsx (hunk 2 above)
Comment (CrazyOptimist) [reply src/features/nft/addNFT/previewStep/PreviewStep.tsx]: I saw a lot of similar comments in other PRs as well, and I'm curios, how do you think:

We will replace all data anyway during api integration. Isn't moving hardcoded value from html to a variable just a extra work?

File: src/common/components/AnimatedModal/AnimatedModal.tsx (hunk 3 above)
Comment (CrazyOptimist) [reply src/common/components/AnimatedModal/AnimatedModal.tsx]: opacity-100 is a Tailwind class

File: src/common/components/Icons/Dots.tsx (hunk 1 above)
Comment (CrazyOptimist) [reply src/common/components/Icons/Dots.tsx]: Fixed

File: src/features/nft/addNFT/uploadStep/UploadingCircle.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  const initialPercent = isReady ? 100 : 0
+
+  return (
+    <div
+      className={`pt-16 flex flex-col items-center bg-center ${style.background}`}
+    >
+      <div className='rounded-full bg-white p-1 relative text-gray-77 mb-5'>
+        <svg
+          viewBox='0 0 100 100'
+          className={`transform -rotate-90 ${style.circle}`}
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply src/features/nft/addNFT/uploadStep/UploadingCircle.tsx:76]: done

File: src/features/nft/addNFT/uploadStep/UploadCircle.module.css [hunk 5]
```diff
@@ -0,0 +1,8 @@
+.background {
+  background-image: url('../../../../common/assets/images/add-nft-upload-background.png');
+}
+
+.circle {
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply src/features/nft/addNFT/uploadStep/UploadCircle.module.css:5]: done

File: src/common/components/AnimatedModal/AnimatedModal.tsx (hunk 3 above)
Comment (reacteer) [reply src/common/components/AnimatedModal/AnimatedModal.tsx]: @CrazyOptimist , Sorry, I had to say "className". We can put this in className, I think.

File: src/features/nft/addNFT/submitStep/SubmitStep.tsx [hunk 6]
```diff
@@ -1,41 +1,98 @@
...
+            <div className='space-y-14px'>
+              <InfoPair title='Title' value='Diamonds in the Sky' />
+              <InfoPair title='Category' value='Illustration' />
+              <InfoPair title='Collection' value='Angels' />
+              <InfoPair title='Copies' value='1 of 1' />
+              <InfoPair title='Compensation' value='Outright' />
+              <InfoPair title='External profile' value='diamondsinthesky.com' />
+              <div className='flex items-center'>
+                <div className='text-gray-71 mr-3'>Green</div>
+                <Toggle toggleHandler={value => console.log(value)} />
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply src/features/nft/addNFT/submitStep/SubmitStep.tsx:63]: Removed, toggleHandler was required before

File: src/features/nft/addNFT/submitStep/SubmitStep.tsx [hunk 7]
```diff
@@ -1,41 +1,98 @@
...
-            Submit and proceed to fee payment
-          </Button>
+      leftColumnContent={
+        <div className='flex-center'>
+          <div className='relative flex-center'>
+            <FullScreenButton onClick={toggleFullScreen} />
+            <img
+              src={image}
+              className='rounded'
+              style={{ maxWidth: '320px', maxHeight: '410px' }}
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply src/features/nft/addNFT/submitStep/SubmitStep.tsx]: done

File: src/features/nft/addNFT/modalLayout/ModalLayout.tsx [hunk 8]
```diff
@@ -44,9 +55,18 @@ export default function CroppingStep({
...
       </div>
-      <div className='flex space-x-7'>
-        <div className={style.leftColumn}>{leftColumnContent}</div>
-        <div className={style.rightColumn}>{rightColumnContent}</div>
+      <div className={cn('flex space-x-7 flex-grow', contentClass)}>
+        <div className={`relative ${style.leftColumn}`}>
+          {leftColumnContent}
+        </div>
+        <div
+          className={style.rightColumn}
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply src/features/nft/addNFT/modalLayout/ModalLayout.tsx]: done

File: src/features/nft/addNFT/AddNFT.module.css [hunk 9]
```diff
@@ -0,0 +1,11 @@
+.modalHeight {
+  height: 555px;
+}
+
+.leftColumn {
+  width: 320px;
+}
+
+.rightColumn {
+  width: 349px;
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply src/features/nft/addNFT/AddNFT.module.css]: done

File: src/features/nft/addNFT/AddNFT.module.css [hunk 10]
```diff
@@ -0,0 +1,11 @@
+.modalHeight {
+  height: 555px;
+}
+
+.leftColumn {
+  width: 320px;
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply src/features/nft/addNFT/AddNFT.module.css]: done

File: src/features/nft/addNFT/AddNFT.module.css [hunk 11]
```diff
@@ -0,0 +1,11 @@
+.modalHeight {
+  height: 555px;
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply src/features/nft/addNFT/AddNFT.module.css]: done

File: src/common/components/Form/TextArea.tsx [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+}: TTextAreaProps): JSX.Element {
+  const textAreaProps = {
+    ...props,
+    className: textAreaClassName,
+  }
+
+  const onChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    const el = e.target
+    el.style.height = ''
+    el.style.height = `${e.target.scrollHeight}px`
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply src/common/components/Form/TextArea.tsx:34]: added comment explaining why it is needed

File: package.json [hunk 13]
```diff
@@ -111,7 +115,8 @@
     "underscore": "1.9.2",
     "update-electron-app": "2.0.1",
     "uuid": "8.3.2",
-    "ws": "7.2.1"
+    "ws": "7.2.1",
+    "yup": "^0.32.9"
```
[the remark this answers is not in the document]
Comment (CrazyOptimist) [reply package.json]: fixed

**Review by reacteer — APPROVED** [review]

**Review by Dicklesworthstone — APPROVED** [review]

**Pull request merged**